src/screens/BrandName.js

```diff
@@ -0,0 +1,148 @@
+import { useNavigation } from "@react-navigation/native";
+import React, { useState, useEffect } from "react";
+import {
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  Pressable,
+  View,
+  TextInput,
+  FlatList,
+} from "react-native";
+
+import BrandNameIcon from "../../assets/SVG/brandNameIcon";
+import CloseIcon from "../../assets/SVG/closeIcon";
+import { COLOURS } from "../constants";
+import brandData from "../json/brand.json";
+
+const BrandNameScreen = () => {
+  const navigation = useNavigation();
+  const [searchQuery, setSearchQuery] = useState("");
+  const [filteredBrands, setFilteredBrands] = useState([]);
+  const [selectedBrand, setSelectedBrand] = useState(null);
+  const handleGoBack = () => {
+    navigation.goBack();
+  };
+  const handleAddNewBrand = () => {
+    if (selectedBrand || searchQuery.trim() !== "") {
+      const brandName = selectedBrand?.name || searchQuery.trim();
```

Comment:
since you have checked selectedBrand truthy above, no need to check the selectedBrand truthy again with the ternary operator

Reply:
Done